### DISCUSSION TITLE
2FA with password and email

### DISCUSSION THREAD

**Discussion by b-tsao** [community/community#70046] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I lost my authentication app for my account with 2FA enabled.

I was able to receive an email notifying me the password I input was correct but login was rejected since my authentication code was input incorrectly.

I tried going through recovery and was able to get to this page
<img width="409" alt="image" src="https://github.com/community/community/assets/8595654/099245fd-7f8b-44f4-8185-b4e36e7cc540">

But none of those apply since my computer has changed since I last logged into this account, I didn't set up SSH keys or personal access token.

How is it that I can't recover an account to which I have
1. my correct password
2. my email which I provided a password recovery token to the recovery process

Can I recover my account with these information?
If I can't, how is it that I'm unable to recover my account with 2 factors which I've proven to have?

The limited options to recover despite having the above is mind boggling.

**Comment by DJLink** [comment]
So your recover token is not working?

I'm assuming you did the steps like this page, in which case having a recovery token should work.  Unlinking email (like your screenshot) should be last resort as that will block access to your account forever.

![image](https://github.com/community/community/assets/423769/cc9d8241-08e8-46ac-bb0e-2e17b1f791fa)

https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/recovering-your-account-if-you-lose-your-2fa-credentials

**Reply by b-tsao** [reply]
Thank you for replying @DJLink

I changed my phone and laptop since I last logged into that account and must have forgotten to transfer the authenticator app as I only have Google authenticator for my other authentication needs on my current phone. Unfortunately there's no key for Github that I can find and thus I can't get the authentication recovery code.

Instead I clicked 2FA account recovery
<img width="455" alt="image" src="https://github.com/community/community/assets/8595654/ab0bd2c0-9a75-4984-9c4c-98b83edfd33d">
<img width="619" alt="image" src="https://github.com/community/community/assets/8595654/0b9fb0ee-c26f-45c1-8b2b-92e86e99bcba">

I was able to receive the email and input the one-time password

<img width="507" alt="image" src="https://github.com/community/community/assets/8595654/ae10b7f9-9267-4934-9028-7e1e7ad54959">

Prior to this I did get this email
<img width="1042" alt="image" src="https://github.com/community/community/assets/8595654/1ad72d78-1566-412c-a16e-1930e2c4fe5c">

Which indicates I did have the correct password.

As mentioned above I changed my laptop as well and since it indicates I did not set up any SSH keys or personal access tokens for the account I'm trying to recover, I'm unable to use any of the available options for step 2.

As you mentioned I really do not want to unlink my email. I believe I do have 2 factors of authentication (original password and email) and would really like to recover this account.

**Comment by DNin01** [comment]
Do you remember saving separate recovery codes when you set up 2FA? They're the little one-time 10-digit codes and they look like this:
`0b92e-6b1db`

If you have access to those, you can recover your account by typing one of them into the "Recovery code" field without even having to verify your email. You might have downloaded them, printed them, or saved them in a password manager.

**Comment by DNin01** [comment]
> How is it that I can't recover an account to which I have
> 
> 1. my correct password
> 2. my email which I provided a password recovery token to the recovery process

Here's what GitHub said, this may answer your question:

> ### Why isn’t email-based authentication an option?
> The account's email address is already used for password reset, which is a form of account recovery. If an attacker has access to an email inbox, they can reset the password for an account and pass the email device verification check, reducing the account's protection to a single factor (email inbox access). We require a second factor to prevent this scenario, meaning that second factor must be distinct from your email inbox. When you enable 2FA, we will no longer perform email verification on login.

**Comment by b-tsao** [comment]
Thank you for replying @DNin01 

Unfortunately I do not remember saving it. I looked in all my notes and could not find it.

Am I royally screwed from recovering my account?

**Comment by AwokeKnowing** [comment]
I recommend everyone leave github

They are now just become bullies and another Microsoft app. 

They are unfit to host the open source community's work. Now they don't even let you decide how to authenticate. they force you to tie your account to SMS or some app. 

Move on.  

Github has lost it's way.Forcing 2fa against dev's will is EXACTLY the kind of behavior that made MS infamous. Why kill github? why don't you get it? MAKE 2FA OPTIONAL
